pkg/lib/serverTime.js

```diff
@@ -258,41 +268,34 @@ export function ServerTime() {
         return status;
     };
 
-    const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
-                                                { superuser: "try" });
-
-    self.get_custom_ntp = function () {
-        /* We only support editing the configuration of
-         * systemd-timesyncd, by dropping a file into
-         * /etc/systemd/timesyncd.conf.d.  We assume that timesyncd is
-         * used when:
-         *
-         * - systemd-timedated is answering for
-         *   org.freedesktop.timedate1 as opposed to, say, timedatex.
-         *
-         * - systemd-timesyncd is enabled (false if chrony is being used)
-         *
-         * The better alternative would be to have an API in
-         * o.fd.timedate1 for managing the list of NTP server
-         * candidates.
-         */
-        const result = {
-            supported: false,
-            enabled: false,
-            servers: []
-        };
+    function service_available(s) {
+        return s.exists && s.unit.LoadState != "masked";
+    }
 
-        if (!timedate1_service.exists || timedate1_service.unit.Id !== "systemd-timedated.service") {
-            console.log("systemd-timedated not in use, ntp server configuration not supported");
-            return Promise.resolve(result);
+    function get_timesync_backend() {
+        // We assume chronyd > ntpd > timesyncd.  If the assumption
+        // turns out to be wrong, we will tell the user afterwards.
+        if (service_available(chronyd_service)) {
+            return "chronyd";
+        } else if (service_available(ntpd_service)) {
```

Comment:
This is about second-guessing the content of ntp-units.d directories in the OSes that we support.  If chronyd is not installed but ntpsec is, then (without checking for ntpd) Cockpit would assume that systemd-timesyncd is active.

We don't have to second-guess if we look into the ntp-units.d directories explicitly.  But even that is just a (much) better guess, since systemd-timedated can also be configured via its environment.

The problem with checking which service is active is that we need to know which service to configure also while NTP is off.

Reply:
I'll have a go at parsing ntp-units.d with some Python, can't be that hard.

Reply:
@mvollmer : TBH this PR is already much more complex than it should be. I propose to completely ignore ntpd for now, and perhaps consider that in a follow-up, or if/when anyone cares.